src/main/java/com/snowflake/kafka/connector/internal/ingestsdk/KcStreamingIngestClient.java

```diff
@@ -0,0 +1,158 @@
+public class KcStreamingIngestClient {
+  private static final String STREAMING_CLIENT_PREFIX_NAME = "KC_CLIENT";
+  private LoggerHandler LOGGER = new LoggerHandler(this.getClass().getName());
+
+  private final SnowflakeStreamingIngestClient client;
+
+  /**
+   * Builds the client's name based on the kc instance and expected client id
+   *
+   * @param kcInstanceId the kafka connector instance id
+   * @param clientId the client id
+   * @return the client's name as 'KC_CLIENT_kcInstanceId_clientId'
+   */
+  public static String buildStreamingIngestClientName(String kcInstanceId, int clientId) {
+    return Utils.formatString("{}_{}_{}", STREAMING_CLIENT_PREFIX_NAME, kcInstanceId, clientId);
+  }
+
+  // TESTING ONLY - inject the client
+  @VisibleForTesting
+  public KcStreamingIngestClient(SnowflakeStreamingIngestClient client) {
+    this.client = client;
+  }
+
+  /**
+   * Creates a streaming client from the given properties and requested name. Validates the
+   * requested client is not null and has the correct name Any exceptions will be passed up, a
+   * sfexception will be converted to a connectexception
+   *
+   * @param streamingClientProps the properties for the client
+   * @param clientName the client name to uniquely identify the client
+   */
+  protected KcStreamingIngestClient(Properties streamingClientProps, String clientName) {
+    try {
+      LOGGER.info("Creating Streaming Client: {}", clientName);
+      this.client =
+          SnowflakeStreamingIngestClientFactory.builder(clientName)
+              .setProperties(streamingClientProps)
+              .build();
+
+      assert this.client != null; // client is final, so never need to do another null check
+      assert this.client.getName().equals(clientName);
+    } catch (SFException ex) {
+      throw new ConnectException(ex);
+    }
+  }
+
+  /**
+   * Creates an ingest sdk OpenChannelRequest and opens the client's channel No exception handling
+   * done, all exceptions will be passed through
+   *
+   * @param channelName the name of the channel to open
+   * @param config config to get the database and schema names for the channel
+   * @param tableName table name of the channel
+   * @return the opened channel
+   */
+  public SnowflakeStreamingIngestChannel openChannel(
+      String channelName, Map<String, String> config, String tableName) {
+    OpenChannelRequest channelRequest =
+        OpenChannelRequest.builder(channelName)
+            .setDBName(config.get(Utils.SF_DATABASE))
+            .setSchemaName(config.get(Utils.SF_SCHEMA))
+            .setTableName(tableName)
+            .setOnErrorOption(OpenChannelRequest.OnErrorOption.CONTINUE)
+            .build();
+    LOGGER.info("Opening a channel with name:{} for table name:{}", channelName, tableName);
+
+    return this.client.openChannel(channelRequest);
+  }
+
+  /**
+   * TODO @rcheng: should add retry here and create() even if sdk retries bc network/load issues?
```

Comment:
I think we should add retries in kafka because there are also failure scenarios in the open/close here

Continuing from @sfc-gh-tzhang comment
https://github.com/snowflakedb/snowflake-kafka-connector/pull/528#discussion_r1091205883

Side note: is there a way move a comment from an 'outdated' file to the current file? Getting tired of having to copy-paste my comments

Reply:
You can use `{@link }` to reference the outdated file in the Javadoc here.

Reply:
When you say, "add retries", do you mean here or in a method that calls this?

Reply:
Add retries here, so the rest of the code can be confident in knowing any KcStreamingIngestClient it uses wont have transient issues